release/ray_release/command_runner/anyscale_job_runner.py

```diff
@@ -0,0 +1,265 @@
+                "could not have been provisioned or Ray was stopped):"
+                f"\n{error}\n"
+            )
+
+        # First try to obtain the output.json from S3.
+        # If that fails, try logs.
+        try:
+            output_json = self.fetch_output()
+        except Exception:
+            logger.exception("Exception when obtaining output from S3.")
+            try:
+                logs = self.get_last_logs()
+                output_json = re.search(r"### JSON \|([^\|]*)\| ###", logs)
+                output_json = json.loads(output_json.group(1))
+            except Exception:
+                output_json = None
+
+        workload_status_code = None
+        if output_json:
+            logger.info(f"Output: {output_json}")
+            workload_status_code = output_json["return_code"]
+            workload_time_taken = output_json["workload_time_taken"]
+            prepare_return_codes = output_json["prepare_return_codes"]
+            last_prepare_time_taken = output_json["last_prepare_time_taken"]
+
+            if prepare_return_codes and prepare_return_codes[-1] != 0:
+                if prepare_return_codes[-1] == TIMEOUT_RETURN_CODE:
+                    raise PrepareCommandTimeout(
+                        "Prepare command timed out after "
+                        f"{last_prepare_time_taken} seconds."
+                    )
+                raise PrepareCommandError(
+                    f"Prepare command '{self.prepare_commands[-1]}' returned "
+                    f"non-success status: {prepare_return_codes[-1]} with error:"
+                    f"\n{error}\n"
+                )
+        else:
+            raise JobNoLogsError("Could not obtain logs for the job.")
+
+        if workload_status_code == TIMEOUT_RETURN_CODE:
+            if is_long_running:
+                # Expected - treat as success.
+                return
+
+            raise TestCommandTimeout(
+                f"Command timed out after {workload_time_taken} seconds."
+            )
+
+        if workload_status_code is not None and workload_status_code != 0:
+            raise TestCommandError(
+                f"Command returned non-success status: {workload_status_code} with "
+                f"error:\n{error}\n"
+            )
+
+        if job_status_code == -1:
+            raise JobStartupFailed(
+                "Job returned non-success state: 'OUT_OF_RETRIES' "
+                "(command has not been ran or no logs could have been obtained) "
+                f"with error:\n{error}\n"
+            )
+
+    @property
+    def command_env(self):
+        env = super().command_env
+        # Make sure we don't buffer stdout so we don't lose any logs.
+        env["PYTHONUNBUFFERED"] = "1"
+        return env
+
+    def run_command(
+        self,
+        command: str,
+        env: Optional[Dict] = None,
+        timeout: float = 3600.0,
+        is_long_running: bool = False,
```

Comment:
Can we use a flag that describes the behavior? Otherwise this leaks a release test concept ("long running") into a generic implementation

```suggestion
        raise_on_timeout: bool = True,
```